upup/pkg/fi/cloudup/urls.go

```diff
@@ -118,39 +119,44 @@ func SetKopsAssetsLocations(assetsBuilder *assets.AssetBuilder) error {
 }
 
 // NodeUpAsset returns the asset for where nodeup should be downloaded
-func NodeUpAsset(assetsBuilder *assets.AssetBuilder) (*MirroredAsset, error) {
-	// Avoid repeated logging
-	if nodeUpAsset != nil {
+func NodeUpAsset(assetsBuilder *assets.AssetBuilder, arch architectures.Architecture) (*MirroredAsset, error) {
+	if nodeUpAsset == nil {
+		nodeUpAsset = make(map[architectures.Architecture]*MirroredAsset)
+	} else if nodeUpAsset[arch] != nil {
 		// Avoid repeated logging
-		klog.V(8).Infof("Using cached nodeup location: %v", nodeUpAsset.Locations)
-		return nodeUpAsset, nil
+		klog.V(8).Infof("Using cached nodeup location: %v", nodeUpAsset[arch].Locations)
+		return nodeUpAsset[arch], nil
+	}
+	// Use multi-arch env var, but fall back to well known env var
+	env := os.Getenv(fmt.Sprintf("NODEUP_URL_%s", strings.ToUpper(string(arch))))
+	if env == "" {
+		env = os.Getenv("NODEUP_URL")
```

Comment:
I was thinking we'd only fall back to this for the amd64 arch.

Reply:
I think it doesn't really matter much. If someone want to use if for a full ARM64 cluster, I don't have anything against it. I personally prefer the `KOPS_BASE_URL` version.